### PR TITLE
refactor: modifyPosition

### PR DESCRIPTION
**Pull request by notJoon**
# Description

**Merge This PR After #400 Is Completely Merged**

Refactored `modifyPosition` function
- removed `NilToZero` converter, because those lines are meaningless if we defined `amount0`, `amount1` value with `i256.Zero`.

### CODE REVIEW

**Comment by notJoon** [comment]
PR #400 was merged. So, changed it to review status.

File: pool/type.gno [hunk 1]
```diff
@@ -171,7 +172,7 @@ func (t *TickInfo) init() {
 
 type Ticks map[int32]TickInfo          // tick => TickInfo
 type TickBitmaps map[int16]*u256.Uint  // tick(wordPos) => bitmap(tickWord ^ mask)
-type Positions map[string]PositionInfo // positionKey => PositionInfo
+type Positions map[string]PositionInfo // positionKey => PositionInfo
```
Comment (r3v4s) [inline pool/type.gno:175]: Aren't we supposed to replace all map to avl?
Since you imported avl from #L8 but doesn't using it. Maybe its WIP??

**Review by r3v4s — COMMENTED** [review]
code looks good, but left one comment for map -> avl

File: pool/type.gno (hunk 1 above)
Comment (notJoon) [reply pool/type.gno:175]: All the PRs are tangled together, I think avl must be apply last.

**Review by onlyhyde — APPROVED** [review]

**Pull request merged**